src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -418,6 +421,17 @@ protected function sendEarlyHints(iterable $links, Response $response = null): R
         $response ??= new Response();
 
         $populatedLinks = [];
+
+        if ($preloadJavaScriptModules) {
+            if (!$this->container->has('importmap.manager')) {
+                throw new \LogicException('You cannot use the JavaScript modules method if the ImportMap component is not available. Try running "composer require symfony/import-map".');
+            }
+
+            foreach ($this->container->get('importmap.manager')->getModulesToPreload() as $url) {
+                $populatedLinks[] = new Link('modulepreload', $url);
```

Comment:
what about adding an attribute that would be added to the controller also? (could be for a followup PR)